### PR TITLE
Move NS client to separate package under NS API

### PR DESCRIPTION
**Pull request by dynco-nym**
- moving NS API client code to separate package outside NS Agent makes the code cleaner & more modular
- NS agent now imports NS API client that it uses
- no functionality change in NS API or NS agent

- - -
This change is [Reviewable](https://reviewable.io/reviews/nymtech/nym/5171)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -164,6 +163,7 @@ default-members = [
     "nym-data-observatory",
     "nym-node",
     "nym-node-status-api",
+    "nym-node-status-api/client",
     "nym-validator-rewarder",
     "nym-node-status-api",
```
Comment (octol) [inline Cargo.toml]: Duplicate?

Comment (dynco-nym) [reply Cargo.toml]: corrected

File: Cargo.toml [hunk 2]
```diff
@@ -126,8 +125,8 @@ members = [
     "nym-node",
     "nym-node/nym-node-http-api",
     "nym-node/nym-node-requests",
-    "nym-node-status-api",
-    "nym-node-status-agent",
+    "nym-node-status-api/nym-node-status-agent",
+    "nym-node-status-api/nym-node-status-api",
```
Comment (octol) [inline Cargo.toml:129]: Looks like `nym-node-status-client` is missing from the workspace

File: nym-node-status-api/nym-node-status-agent/Cargo.toml [hunk 3]
```diff
@@ -1,7 +1,6 @@
 # Copyright 2024 - Nym Technologies SA <contact@nymtech.net>
 # SPDX-License-Identifier: Apache-2.0
 
-
 [package]
 name = "nym-node-status-agent"
 version = "1.0.0-rc.1"
```
Comment (octol) [inline nym-node-status-api/nym-node-status-agent/Cargo.toml:6]: Just a side question. What is the versioning and release process for these?

**Review by octol — APPROVED** [review]

File: Cargo.toml (hunk 2 above)
Comment (dynco-nym) [reply Cargo.toml:129]: added

File: nym-node-status-api/nym-node-status-agent/Cargo.toml (hunk 3 above)
Comment (dynco-nym) [reply nym-node-status-api/nym-node-status-agent/Cargo.toml:6]: discussed, closing

**Pull request merged**